src/controllers/redeploy.go

```diff
@@ -4,13 +4,15 @@ import (
 	"github.com/devclub-iitd/DeployBot/src/history"
 )
 
-func redeploy(callbackID string, data map[string]interface{}) {
-	stop(callbackID, data)
-	state, _ := history.GetState(data["git_repo"].(string))
-	if state.Status == "stopped" {
-		data["subdomain"] = state.Subdomain
-		data["access"] = state.Access
-		data["server_name"] = state.Server
-		deploy(callbackID, data)
+func redeploy(params *DeployAction) {
+	state, _ := history.GetState(params.data["git_repo"].(string))
+
+	params.data["subdomain"] = state.Subdomain
+	params.data["access"] = state.Access
+	params.data["server_name"] = state.Server
+
+	if state.Status != "stopped" {
+		params.command = "redeploy"
```

Comment:
isn't the `command` already being set to `redeploy` [here](https://github.com/devclub-iitd/DeployBot/blob/67124a9c35235aadd7a3db209706dc97f3e37cb2/src/controllers/handlers.go#L33)